app/blueprints/assessments/routes.py

```diff
@@ -704,6 +718,65 @@ def application(application_id):
     )
 
 
+@assessment_bp.route("/activity_trail/<application_id>", methods=["GET"])
+@check_access_application_id
+def activity_trail(application_id: str):
+    state = get_state_for_tasklist_banner(application_id)
+
+    # There is a better way of doing it by moving
+    # all activity related logics to an endpoint in the
+    # assessment store and write up a query to fetch all the information.
+
+    # ALL FLAGS
+    flags_list = get_flags(application_id)
+    _flags = Flags.from_list(flags_list)
+
+    # ALL COMMENTS
+    comments_list = get_comments(application_id)
+    all_comments = Comments.from_list(comments_list)
+
+    # ALL SCORES
+    scores = get_score_and_justification(
+        application_id=application_id, score_history=True
+    )
+    all_scores = Scores.from_list(scores)
+
+    # ALL TAGS
+    tags = get_all_associated_tags_for_application(application_id)
+    _tags = AssociatedTags.from_associated_tags_list(tags)
+
+    # TODO: Get current assessment status
+
+    # TODO: GET workflow STATUSES
+
+    # Add search box and checkbox filters
+    available_filters = ["All activity", "Comments", "Score", "Flags", "Tags"]
+    search_form = SearchForm(request.form)
+    checkbox_form = CheckboxForm(request.form)
+
+    # Filter all activities
+    search_keyword = request.args.get("search")
+    checkbox_filters = request.args.getlist("filter")
+    all_activities = all_scores + all_comments + _tags + _flags
```

Comment:
Can we use the same convention rather than `all_...` & `_...`?

Reply:
Yes! will do 😄 . Didnt realise